src/inference/elbo.ad.js

```diff
@@ -303,7 +303,12 @@ module.exports = function(env) {
       return k(s);
     },
 
-    mapDataFetch: function(data, batchSize, address) {
+    mapDataFetch: function(data, opts, address) {
+
+      var batchSize = opts.batchSize !== undefined ? opts.batchSize : data.length;
+      if (batchSize < 0 || batchSize > data.length) {
```

Comment:
It looked to me like we're currently supporting `batchSize` 0 (due to checking `batchSize < 0` instead of `batchSize <= 0`), but I didn't see the utility in it / thought that it would more likely result from a mistake than from an intentional choice. I don't think we should add it if it's not supported right now.